front/src/utils/strings.ts

```diff
@@ -66,3 +68,28 @@ export function convertInputStringToNumber(str: string) {
 export function formatUicToCi(uic: number) {
   return uic.toString().replace(/^87/, '');
 }
+
+/**
+ * Given a distance in millimeters, returns a human readable string.
+ * Exemples:
+ * - 927520 -> 927m
+ * - 1927520 -> 1.92Km
+ *
+ * @param distance the distance in millimeters
+ * @returns A human readable string of the distance
+ */
+export function distanceToHumanReadable(distance: number): string {
+  let label = `${distance}mm`;
+  if (!distance) return '0';
+
+  const meters = distance / 1000;
+  if (Math.floor(meters) > 0) {
+    label = `${round(meters, 0)}m`;
+  }
+
+  const kms = distance / 1000 / 1000;
+  if (Math.floor(kms) > 0) {
+    label = `${round(kms, 2)}Km`;
+  }
+  return label;
+}
```

Comment:
It's not a heavy function, but maybe we can optimize it like that instead of sometimes passing two times in the condition by checking km first ? (I also added some utils we have for conversions)

```suggestion
export function distanceToHumanReadable(distance: number): string {
  if (!distance) return '0';

  const kms = mmToKm(distance);
  if (Math.floor(kms) > 0) return `${round(kms, 2)}Km`;

  const meters = mmToM(distance);
  if (Math.floor(meters) > 0) return `${round(meters, 0)}m`;

  return `${distance}mm`;
}
```